adding Nathan's SSO/Custom IDP docs (#625)
chainguard-dev/edu: content/chainguard/chainguard-enforce/authentication/example-idps/azure-ad/index.md

The Chainguard platform supports Single Sign-on (SSO) authentication for users. By default, users can log in with GitHub, Gitlab and Google, but SSO support allows users to bring their own identity provider for authentication. 

This guide outlines how to create an Azure Active Directory (AD) Application and integrate it with Chainguard Enforce. After completing this guide, you'll be able to log in to Chainguard Enforce using Azure AD and will no longer be limited to the default SSO options.

## Prerequisites

To complete this guide, you will need the following.

* `chainctl` installed on your system. Follow our guide on [How To Install `chainctl`](/chainguard/chainguard-enforce/how-to-install-chainctl/) if you don't already have this installed.
* An Azure account you can use to set up an Active Directory Application. 

## Create an Azure Active Directory Application

To integrate the Azure AD identity provider with the Chainguard platform, log in to [Azure](https://azure.microsoft.com) and navigate to the Azure Active Directory console in the Azure portal. 

![Screenshot of the Azure portal's sidebar menu, with Azure Active Directory highlighted in a red circle.](aad-1.png)

> There, select the **App registrations** tab and click **New Registration**.

Author: nice catch!

Finally, navigate to the **Certificates & secrets** tab to create a client secret to authenticate the Chainguard platform to Azure Active Directory. Select **New client secret** to add a client secret. In the resulting modal window, add a description and set an expiration date.

![Screenshot showing the Certificates & secrets landing page with the Add a client secret screen opened. The Certificates & secrets tab is highlighted in a red circle, as is the New client secret button.](aad-6.png)

Finally, take note of the client secret “Value” that is created. You'll need this to configure the Chainguard platform to use this Azure Active Directory application.

![Screenshot showing the client secrets tab, with the Value highlighted in a red circle.](aad-7.png)

## Configuring Chainguard to use Azure Active Directory

> To configure Chainguard make a note of the following details from your Azure Active Directory application:

Reviewer: ```suggestion
To configure Chainguard, make a note of the following details from your Azure Active Directory application:
```

* **Application (client) Id**: This can be found on the **Overview** tab of the Chainguard AD application.
* **Client Secret**: You noted this down when you set up the  clientsecret in the previous step.
* **Directory (tenant) Id**: This can also be found on the **Overview** tab of the Chainguard AD application.

![Screenshot of the Azure AD Overview tab showing the Essentials information. THe Application (client) ID and the Directory (tenant) ID are both highlighted in red circles.](aad-8.png)

> Next, log in to Chaingaurd with `chainctl`, using a social provider like Google, Github, or Gitlab to bootstrap your account.

Reviewer: ```suggestion
Next, log in to Chainguard with `chainctl`, using an OIDC provider like Google, GitHub, or GitLab to bootstrap your account.
```

```sh
chainctl auth login
```

> Note that this bootstrap account can be used as a ["break glass" account](/chainguard/chainguard-enforce/authentication/custom-idps/#break-glass-accounts) (that is, a backup account you can use to log in if you ever lose access to your primary account). However, if you prefer to remove this rolebinding after configuring the custom IDP, you may also do so.

Reviewer: "break glass" spelling consistency here too
